core/src/main/java/com/linecorp/armeria/common/util/MoreSystem.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.util;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.net.IDN;
+import java.net.InetAddress;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Ascii;
+
+import io.netty.util.internal.PlatformDependent;
+
+/**
+ * Provides utilities for accessing the information about the current system and process.
+ */
+public final class MoreSystem {
```

Comment:
I guess this is following the `More` convention with the `System` class. It does read quite strangely though. Do you see it being more than just providing system information? If not, could just be `SystemInformation` or similar.

Reply:
Asked the naming question at Twitter and got the following:

- `Os`
- `PosixUtil`
- `LocalServerInfo`
- `SystemHosts`
- `CurrentSystem`
- `LocalSystem`
- `Sys`

Perhaps, `SystemInfo`?

Reply:
+1 to `SystemInfo`